src/main/java/uk/gov/hmcts/reform/unspec/service/tasks/handler/CaseEventTaskHandler.java

```diff
@@ -5,32 +5,52 @@
 import org.camunda.bpm.client.task.ExternalTaskHandler;
 import org.camunda.bpm.client.task.ExternalTaskService;
 import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDataContent;
+import uk.gov.hmcts.reform.ccd.client.model.Event;
+import uk.gov.hmcts.reform.ccd.client.model.StartEventResponse;
 import uk.gov.hmcts.reform.unspec.callback.CaseEvent;
+import uk.gov.hmcts.reform.unspec.helpers.CaseDetailsConverter;
 import uk.gov.hmcts.reform.unspec.model.BusinessProcess;
+import uk.gov.hmcts.reform.unspec.model.CaseData;
+import uk.gov.hmcts.reform.unspec.model.ExternalTaskInput;
 import uk.gov.hmcts.reform.unspec.service.CoreCaseDataService;
 
+import java.util.HashMap;
 import java.util.Map;
 
 @RequiredArgsConstructor
 @Component
 public class CaseEventTaskHandler implements ExternalTaskHandler {
 
     private final CoreCaseDataService coreCaseDataService;
+    private final CaseDetailsConverter caseDetailsConverter;
 
     @Override
     public void execute(ExternalTask externalTask, ExternalTaskService externalTaskService) {
         Map<String, Object> allVariables = externalTask.getAllVariables();
-        String ccdId = (String) allVariables.get("CCD_ID");
-        String eventId = (String) allVariables.get("CASE_EVENT");
-
-        coreCaseDataService.triggerEvent(
-            Long.valueOf(ccdId),
-            CaseEvent.valueOf(eventId),
-            Map.of(
-                "businessProcess",
-                BusinessProcess.builder().activityId(externalTask.getActivityId()).build()
-            )
-        );
+        ExternalTaskInput externalTaskInput = caseDetailsConverter.fromMap(allVariables, ExternalTaskInput.class);
```

Comment:
using caseDetailsConverter doesn't seem right here, we aren't converting caseDetails